src/App.js

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js'
+
 
 const App = () => {
+  const localUser = chatMessages[0].sender
+  const remoteUser = chatMessages[1].sender
+
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const onUpdateChat = (updatedChat) => {
+    const messages = chatData.map(chat => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return chat;
+      }
+    });
```

Comment:
Rather than receiving a new entry with it's like status already changed here, rewrite this method to accept only an id. The logic about how to copy the message and change the liked status would be better if it was written in App.js, since the state must have all the information necessary to make a new message. 

```javascript
const onUpdateChat = id => {
const messages = chatData.map(chat => {
      if (chat.id === id){
        return {...chat, liked: !chat.liked};
      } 
      else {
        return chat;
      }
    });